File: Chapitre2/exercice2.2.c
//créer 2 fils
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <errno.h>


int main()
{
    pid_t fils1;
    fils1=fork();
    if( fils1 == -1)
    {
        perror("Erreur lors de la création d'un des processus");
        exit(1);
    }
    if(fils1 == 0)
    {
        printf("Je suis dans le fils1 : moi=%d,père=%d\n",getpid(),getppid());
    }
    else  //dans le père
    {
        pid_t fils2;
        fils2=fork();
        if( fils2 == 0)  //dans le fils2
        {
            printf("Je suis dans le fils2 : moi=%d,père=%d\n",getpid(),getppid());
        }
        else
        {
            printf("Je suis le père: fils1:%d,fils2=%d,moi:%d,mon_père:%d\n", fils1,fils2,getpid(), getppid());
        }
        
    }

    return 0;
}

File: Chapitre2/exercice2.2.py
import os
import sys

def main():
    fils1 = os.fork()
    if fils1 == -1:
        print("Erreur lors de la création d'un des processus", file=sys.stderr)
        sys.exit(1)

    if fils1 == 0:
        print(f"Je suis dans le fils1 : moi={os.getpid()},père={os.getppid()}")
    else:
        fils2 = os.fork()
        if fils2 == 0:
            print(f"Je suis dans le fils2 : moi={os.getpid()},père={os.getppid()}")
        else:
            print(f"Je suis le père: fils1:{fils1},fils2={fils2},moi:{os.getpid()},mon_père:{os.getppid()}")

if __name__ == "__main__":
    main()